dashboard/entry/update.go

```diff
@@ -53,6 +53,25 @@ func (r *ResolverForEntry) UpdateDashboardEntry(ctx context.Context, id int, ent
 			entry.RangeFrom = stats.Range.From
 			entry.RangeTo = stats.Range.To
 		}
+
+		if tag := tagsDuplicates(stats.ExcludeTags, stats.IncludeTags); tag != nil {
+			return nil, fmt.Errorf("tag '%s' is present in both exclude tags and include tags", tag.Key+":"+tag.Value)
+		}
+
+		r.DB.Where("dashboard_entry_id = ?", id).Delete(new(model.DashboardExcludedTag))
```

Comment:
This is potentially dangerous, as the tags are here deleted, but the `r.DB.Save(entry)` at the end may not be executed when a check inbeteen fails, e.g. the tagsExist.

It should use a transaction.

Reply:
This is still open, a transaction should be used.